app/forms/claims/mentor_form.rb

```diff
@@ -19,15 +27,30 @@ def as_form_params
   end
 
   def mentor
-    @mentor ||= MentorBuilder.call(trn:, first_name:, last_name:).becomes(Claims::Mentor)
+    @mentor ||= Claims::MentorBuilder.call(
+      trn:,
+      first_name:,
+      last_name:,
+      date_of_birth:,
+    ).becomes(Claims::Mentor)
   end
 
   def validate_mentor
     if mentor.errors.any?
-      mentor.errors.each { |err| errors.add(:trn, err.message) }
+      mentor.errors.each { |err| errors.add(err.attribute, err.message) }
     end
   end
 
+  def date_of_birth
+    Date.new(
+      attributes["date_of_birth(1i)"].to_i, # year
+      attributes["date_of_birth(2i)"].to_i, # month
+      attributes["date_of_birth(3i)"].to_i, # day
+    )
+  rescue Date::Error
+    nil
+  end
```

Comment:
We use this enough to warrant a helper method.

```ruby
date_attribute :date_of_birth

def self.date_attribute(attr_name)
  define_method(attr_name) do
    Date.new(
      attributes["#{attr_name}(1i)"].to_i, # year
      attributes["#{attr_name}(2i)"].to_i, # month
      attributes["#{attr_name}(3i)"].to_i, # day
    )
  rescue Date::Error
    nil
  end
end
```

\* I see where we use `attributes` now 🤔. Maybe we do `#send`?

Reply:
\* Not a blocker

Reply:
This is quite different now. It will return nil if the values are not integers. This form cannot send dates like `12-12-0000` but the filter form that has a similar method to this can